File: deploy/inference_api.js
//API call for JavaScript backend
async function query(data) {
	const response = await fetch(
		"https://api-inference.huggingface.co/models/AshArya/ITRLTrained",
		{
			headers: { Authorization: "Bearer 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" },
			method: "POST",
			body: JSON.stringify(data),
		}
	);
	const result = await response.json();
	return result;
}

query({"inputs": "The answer to the universe is"}).then((response) => {
	console.log(JSON.stringify(response));
});

File: deploy/inference_api.py
# API call for the model from HuggingFace
import requests

API_URL = "https://api-inference.huggingface.co/models/AshArya/ITRLTrained"
headers = {"Authorization": "Bearer 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"}

def query(payload):
	response = requests.post(API_URL, headers=headers, json=payload)
	return response.json()
	
output = query({
	"inputs": "The answer to the universe is", #The 'inputs' should be replaced by the required query
})

#Ideally, the translated algorithm statement should be assigned to a variable and used within the query() function.
#Any inconsistencies, please message.
